cpp/src/neighbors/brute_force_c.cpp

```diff
@@ -67,8 +67,8 @@ void _search(cuvsResources_t res,
   using queries_mdspan_type   = raft::device_matrix_view<T const, int64_t, QueriesLayoutT>;
   using neighbors_mdspan_type = raft::device_matrix_view<int64_t, int64_t, raft::row_major>;
   using distances_mdspan_type = raft::device_matrix_view<float, int64_t, raft::row_major>;
-  using prefilter_mds_type    = raft::device_vector_view<const uint32_t, int64_t>;
```

Comment:
I think we want to keep the filter immutable, don' we?

Reply:
This change is to be compatible with the `bitset` filter utilized in CAGRA and other algorithms. The immutable is implemented on the `search` API by restricting the filter to be `const cuvs::neighbors::filtering::base_filter`. Please refer to [cagra](https://github.com/rapidsai/cuvs/blob/branch-25.02/cpp/src/neighbors/cagra.cuh#L350), here

Reply:
We are using obj<const type> everywhere in the codebase so we should be doing that here as well. If the CAGRA API is not doing this, it should be.